### ISSUE TITLE
Compile time error for get<T>() if T not in type list?

### ISSUE THREAD

**Issue by joto** [mapbox/variant#24]
Should `get<double>()` generate a compile time error in the following case?

```
variant<int> v = 1;
auto x = v.get<double>()
```
because `get<double>` can never work.

Note that this is different than
```
variant<int, double> v = 1;
auto x = v.get<double>()
```
which can work if there is something else in between the lines that changes `v` into a `double`.

**Comment by artemp** [comment]
fixed in c117592337cc20c238d4ce9f9d8847aff0cd55ab

**Comment by DennisOSRM** [comment]
cool, upgrading
